
Note: This is needed since gtk+3.0 is not compiled with the option :   --enable-wayland-backend
therefore, the gdk/gdkwayland.h is not included in gtk+3.0-dev package.
fcolista@alpinelinux.org

diff '--exclude=.git' -urN a/plugins/common/gsd-device-manager-udev.c b/plugins/common/gsd-device-manager-udev.c
--- a/plugins/common/gsd-device-manager-udev.c	2017-07-22 12:29:49.000000000 -0400
+++ b/plugins/common/gsd-device-manager-udev.c	2017-07-27 05:25:11.728250282 -0400
@@ -24,7 +24,9 @@
 #include <string.h>
 #include <gudev/gudev.h>
 
+#ifdef GDK_WINDOWING_WAYLAND
 #include <gdk/gdkwayland.h>
+#endif
 #include "gsd-device-manager-udev.h"
 
 struct _GsdUdevDeviceManager
@@ -233,6 +235,7 @@
 gsd_udev_device_manager_lookup_device (GsdDeviceManager *manager,
 				       GdkDevice	*gdk_device)
 {
+#ifdef GDK_WINDOWING_WAYLAND
 	const gchar *node_path;
 	GHashTableIter iter;
 	GsdDevice *device;
@@ -249,6 +252,7 @@
 			return device;
 		}
 	}
+#endif
 
 	return NULL;
 }
